crates/orchestrator/src/tests/database/mod.rs

```diff
@@ -1,7 +1,63 @@
+use crate::config::{config, TestConfigBuilder};
+use crate::jobs::types::{ExternalId, JobItem, JobStatus, JobType};
+use crate::tests::common::drop_database;
 use rstest::*;
+use uuid::Uuid;
 
 #[rstest]
 #[tokio::test]
-async fn test_database() {
-    // TODO: write test case
+async fn test_database_connection() -> color_eyre::Result<()> {
+    TestConfigBuilder::new().build().await;
+    Ok(())
+}
+
+/// Tests for `create_job` operation in database trait.
+/// Creates 3 jobs and asserts them.
+#[rstest]
+#[tokio::test]
+async fn test_database_create_job() -> color_eyre::Result<()> {
+    TestConfigBuilder::new().build().await;
+
+    drop_database().await.unwrap();
```

Comment:
Use a fixture to start with a fresh DB and an already built config.

Reply:
Valid,
We had used `fixture` in the initial stages of tests but we soon realised that we require more customisability like mocking only `da_client` while writing tests, hence we went with the [Builder Design Pattern](https://rust-unofficial.github.io/patterns/patterns/creational/builder.html) to achieve the same.

Moreover : 
- `drop_database` is an async fn, we didn't find this to be a valid enough reason to implement `Drop` on Config.

Proposed Solution : 
- we add `drop_database` as a part of `TestConfigBuilder.build()` so that we drop the database once before each test-case runs, which gives same result as dropping database after each test-case.

WDYT ?

Reply:
Several things to say here.

1. Regarding when to drop the DB, I believe it's best to drop it when the test starts and not after it finishes. This way, you can inspect the DB to understand what went wrong.
2. Fixture vs manual call: While you're going to have multiple test configs, you're only going to use a few. So defining a few fixtures with the right configuration options on the test builder that return the test config is probably the right thing to do. This enables finer-grained control on when to reset the DB (once for a group of tests? For each test?) depending on what you want.
3. We didn't talk about test concurrency here, but if you let all your tests execute in parallel you have a strong risk of dropping the DB in the middle of another test. Do you have a solution for this?

Reply:
1. Yes so we will do that in the beginning of the tests only.
2. Will have to think about that can you give me an example on this if possible ?
3. We were also discussing about this few days back. We will have to come up on this solution but currently we don't have one yet. We were thinking to make collections test wise or databases for different tests something like that.

Reply:
To add on (3), the major blocker for concurrency would be DB or AWS localstack. We haven't finalized the implementation yet but it shouldn't be difficult to add on the current implementation. As @ocdbytes mentioned, we were looking at mostly using different databases for each test case and different localstack accounts. I think the changes should be localized to the builder itself but the idea was to focus on this more when we see test cases taking >15mins or so. wdyt?